sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala

```diff
@@ -140,7 +141,7 @@ private[hive] object SparkSQLCLIDriver extends Logging {
       // Hadoop-20 and above - we need to augment classpath using hiveconf
       // components.
       // See also: code in ExecDriver.java
-      var loader = conf.getClassLoader
+      var loader = orginalClassLoader
```

Comment:
I think this is a correct fix. Another approach is add the Spark jars to `Utilities.addToClassPath` to make `UDFClassLoader` work:
```scala
val sparkJars = sparkConf.get(org.apache.spark.internal.config.JARS)
if (sparkJars.nonEmpty || StringUtils.isNotBlank(auxJars)) {
  loader = Utilities.addToClassPath(loader, sparkJars.toArray ++ StringUtils.split(auxJars, ","))
}
```

Reply:
cc @juliuszsompolski